iati/home/tests/test_middleware.py

```diff
@@ -0,0 +1,49 @@
+"""A module of unit tests for redirects."""
+import pytest
+from django.conf import settings
+from factories import DocumentFactory
+
+
+@pytest.mark.django_db
+class TestRedirectMiddleware():
+    """Tests for the custom redirect middleware."""
+
+    def create_document(self):
+        """Call the DocumentFactory to save a new document."""
+        return DocumentFactory.create()
+
+    @pytest.mark.parametrize('redirect_mapping', [
+        ('/HoMe/', '/home/'),
+        ('/HOME/', '/home/'),
+        ('/home/', '/en/home/'),
+        ('/104/MiXeDCaSe/', 'http://reference.iatistandard.org/104/MiXeDCaSe/'),
+        ('/105/MiXeDCaSe/', 'http://reference.iatistandard.org/105/MiXeDCaSe/'),
+        ('/201/MiXeDCaSe/', 'http://reference.iatistandard.org/201/MiXeDCaSe/'),
+        ('/202/MiXeDCaSe/', 'http://reference.iatistandard.org/202/MiXeDCaSe/'),
+        ('/203/MiXeDCaSe/', 'http://reference.iatistandard.org/203/MiXeDCaSe/'),
+        ('/activity-standard/MiXeDCaSe/', 'http://reference.iatistandard.org/activity-standard/MiXeDCaSe/'),
+        ('/codelists/MiXeDCaSe/', 'http://reference.iatistandard.org/codelists/MiXeDCaSe/'),
+        ('/developer/MiXeDCaSe/', 'http://reference.iatistandard.org/developer/MiXeDCaSe/'),
+        ('/introduction/MiXeDCaSe/', 'http://reference.iatistandard.org/introduction/MiXeDCaSe/'),
+        ('/namespaces-extensions/MiXeDCaSe/', 'http://reference.iatistandard.org/namespaces-extensions/MiXeDCaSe/'),
+        ('/organisation-identifiers/MiXeDCaSe/', 'http://reference.iatistandard.org/organisation-identifiers/MiXeDCaSe/'),
+        ('/organisation-standard/MiXeDCaSe/', 'http://reference.iatistandard.org/organisation-standard/MiXeDCaSe/'),
+        ('/reference/MiXeDCaSe/', 'http://reference.iatistandard.org/reference/MiXeDCaSe/'),
+        ('/rulesets/MiXeDCaSe/', 'http://reference.iatistandard.org/rulesets/MiXeDCaSe/'),
+        ('/schema/MiXeDCaSe/', 'http://reference.iatistandard.org/schema/MiXeDCaSe/'),
+        ('/upgrades/MiXeDCaSe/', 'http://reference.iatistandard.org/upgrades/MiXeDCaSe/'),
+        (settings.STATIC_URL + "HOME/css/", "/en" + settings.STATIC_URL + "HOME/css/")
+    ])
+    def test_redirect_middleware_default(self, client, redirect_mapping):
```

Comment:
These can be split into more granular tests, for example,

* `test_internal_redirects()`, where `redirect_mapping` will contain internal links
* `test_reference_redirects()`, where `redirect_mapping` will contain the external URLs
* `test_static_redirects()`, where `redirect_mapping` will contain the static file URLs


this makes it simpler to understand which test in particular is failing - it's best to test for one common thing in each test, and in this case, it makes sense to test each inclusion/exclusion rule we defined